File: 7-kyu/naming-files/python/solution.py
def name_file(fmt, nbr, start):
    if nbr <= 0 or not isinstance(nbr, int) or not isinstance(start, int):
        return []
    if fmt.find('<index_no>') == -1:
        return [fmt] * nbr
    ret = []
    for i in range(start, start + nbr, 1):
        fmt_copy = ''.join(ch for ch in fmt)
        fmt_copy = fmt_copy.replace('<index_no>', str(i))
        ret.append(fmt_copy)
    return ret

File: 7-kyu/naming-files/javascript/solution.js
const nameFile = (fmt, nbr, start) => {
  if (nbr <= 0 || !Number.isInteger(nbr) || !Number.isInteger(start)) {
    return [];
  }
  else if (!fmt.includes("<index_no>")) {
    return Array(nbr).fill(fmt);
  }
  const ret = [];
  for (let i = start; i < start + nbr; i++) {
    const next = fmt.replaceAll("<index_no>", `${i}`);
    ret.push(next);
  }
  return ret;
}